dbms/src/Interpreters/Aggregator.cpp

```diff
@@ -1474,45 +1597,68 @@ std::vector<std::unique_ptr<AggregatorMethodInitKeyColumnHelper<Method>>> initAg
 }
 } // namespace
 
-template <typename Method, typename Table>
+template <typename Method, typename Table, bool skip_convert_key>
 void NO_INLINE Aggregator::convertToBlocksImplFinal(
     Method & method,
     Table & data,
+    const Sizes & key_sizes,
     std::vector<std::vector<IColumn *>> && key_columns_vec,
     std::vector<MutableColumns> & final_aggregate_columns_vec,
     Arena * arena) const
 {
     assert(!key_columns_vec.empty());
```

Comment:
If `skip_convert_key` is true, `key_columns_vec` maybe empty?

Reply:
`key_columns_vec` should be emtpy if `skip_convert_key` is true. But currently it's not, because in `prepareBlockAndFill`, it will fill `MutableColumns` even when `convert_key_size` is zero. Check [here](https://github.com/pingcap/tiflash/blob/bab33aa025adcc19b57f783229e63215930a0faa/dbms/src/Interpreters/Aggregator.cpp#L1643).

Maybe I should change it.

Reply:
I think it's better to change it, and keep `key_sizes.size() == key_columns_vec.size()` always true?

Reply:
Add check for `key_sizes.size() == key_columns_vec[i].size()` for all four convert functions.